packages/ra-core/src/form/FormDataConsumer.tsx

```diff
@@ -99,26 +105,24 @@ export const FormDataConsumerView = (props: Props) => {
 };
 
 export default FormDataConsumer;
-
-export interface FormDataConsumerRenderParams {
-    formData: any;
+export interface FormDataConsumerRenderParams<
+    TFieldValues extends FieldValues = FieldValues
+> {
+    formData: TFieldValues;
     scopedFormData?: any;
     getSource?: (source: string) => string;
 }
-
-export type FormDataConsumerRender = (
-    params: FormDataConsumerRenderParams
-) => ReactNode;
-
-interface ConnectedProps {
-    children: FormDataConsumerRender;
+export type FormDataConsumerRender<
+    TFieldValues extends FieldValues = FieldValues
+> = (params: FormDataConsumerRenderParams<TFieldValues>) => ReactNode;
+interface ConnectedProps<TFieldValues extends FieldValues = FieldValues> {
+    children: FormDataConsumerRender<TFieldValues>;
     form?: string;
     record?: any;
     source?: string;
     [key: string]: any;
 }
-
-interface Props extends ConnectedProps {
-    formData: any;
+interface Props<TFieldValues extends FieldValues> extends ConnectedProps {
```

Comment:
@Gabriel-Malenowitch we had the line broken in some cases, i think it's worth going back.

Reply:
I've reviewed it here and couldn't identify the broken lines you say (only from eslint, which broke a few).
This interface (`Props`) has just been improved. Were you talking about her?

![image](https://user-images.githubusercontent.com/56513919/202045481-bfad36ef-f04c-4e11-ae3c-ffc900b5ba2f.png)

Reply:
@Gabriel-Malenowitch I don't know why but it's not inferring optional fields neither null ones.

Reply:
I'm surprised you don't need to do

```suggestion
interface Props<TFieldValues extends FieldValues> extends ConnectedProps<TFieldValues> {
```

or something similar

Reply:
Besides, you removed all the extra lines breaks in the file, which I think hurts its readability. Can you add them back?

Reply:
Nice point guys! https://github.com/marmelab/react-admin/pull/8389/commits/0d771fe7c1fdf075e204921670b31a60dd7cbc46